### PR TITLE
chore: updated changelog to v0.6.0

### PR DESCRIPTION
**Pull request by PacoDw**
## Description

Updating changelog file

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,47 @@
 ## 0.6.0 (Unreleased)
+
+**Implemented enhancements:**
+
+- mongodbatlas\_database\_user can not be imported when they contain dashes "-" in the name [\#179](https://github.com/terraform-providers/terraform-provider-mongodbatlas/issues/179)
```
Comment (themantissa) [inline CHANGELOG.md]: Move this (line 5, issue #179 ) to closed issues

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,47 @@
 ## 0.6.0 (Unreleased)
+
+**Implemented enhancements:**
+
```
Comment (themantissa) [inline CHANGELOG.md:8]: Add the following under "Implemented enhancements" to better highlight the enhancements the team made:

- New parameters about pagination for datasources [\#237](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/237) ([coderGo93](https://github.com/coderGo93))
- Added support for cluster autoscaling attributes [\#233](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/233) ([coderGo93](https://github.com/coderGo93))
- Migrated to new Terraform SDK [\#229](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/229) ([PacoDw](https://github.com/PacoDw))
- Added attribute container\_id to the cluster resource (useful for when a cluster exists before creating a peering connection) [\#208](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/208) ([coderGo93](https://github.com/coderGo93))
- Added attributes to snapshot restore jobs resource and datasources to support continuous cloud backup [\#224](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/224) ([coderGo93](https://github.com/coderGo93))
- General documentation improvements, Guide section add, and Upgrade Guide for 0.6.0 [\#240](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/240) ([themantissa](https://github.com/themantissa))
- General documentation improvements: connection string doc [\#223](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/223) ([themantissa](https://github.com/themantissa))
- General documentation improvements: network peering doc [\#217](https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/217) ([themantissa](https://github.com/themantissa))

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,47 @@
 ## 0.6.0 (Unreleased)
+
```
Comment (themantissa) [inline CHANGELOG.md:2]: Add line before enhancements section:
Recommendation: Before upgrading read the [MongoDB Atlas Provider 0.6.0: Upgrade Guide](https://www.terraform.io/docs/providers/mongodbatlas/guides/0.6.0-upgrade-guide.html)

**Review by themantissa — CHANGES_REQUESTED** [review]
Thank you @PacoDw - just a few things to move around and I wrote up the enhancement PRs into a more descriptive list.

File: CHANGELOG.md (hunk 1 above)
Comment (PacoDw) [reply CHANGELOG.md]: okay

File: CHANGELOG.md (hunk 2 above)
Comment (PacoDw) [reply CHANGELOG.md:8]: done

**Review by themantissa — APPROVED** [review]
LGTM :shipit:

**Review by coderGo93 — APPROVED** [review]
LGTM

**Pull request merged**